Design/DesignDoc.md

```diff
@@ -0,0 +1,225 @@
+# Design Document
+
+## _BudgetMe_ App Design
+
+## 1. Problem Statement
+
+BudgetMe is a highly advanced budgeting/expense tracking app that will enable users to have more control over their money/expenditures.
+Users can currently view their expenses by visiting the associated website/app for their card provider. However, users have requested that their expenditures and budget plans should be presented in a more streamlined, informative manner.
+
+This design document describes the main use cases and functionality of BudgetMe, a new service that implements AWS technologies and will provide users a cleaner, more informative means to view expenditures and budget goals.
+It is designed to interact with a front end component (which allows users to view expenditure information based on assigned tags and budget information based on their set goal).
+
+
+## 2. Top Questions to Resolve in Review
+
+_List the most important questions you have about your design, or things that you are still debating internally that you might like help working through._
+
+1. Can we search expenses by date? 
+2. Would date need to be a string rather than a ZoneDateTime?
+3. Would status need to be string rather than boolean?
+4. Does the budget model need more variables? 
+
+
+## 3. Use Cases
+
+_This is where we work backwards from the customer and define what our customers would like to do (and why). You may also include use cases for yourselves (as developers), or for the organization providing the product to customers._
```

Comment:
Don't forget to remove/cleanup all of the instructions like this throughout the doc.